client/components/Map.tsx

```diff
@@ -1,53 +1,85 @@
 import React, { useState, useEffect } from "react";
-import MapView from 'react-native-maps';
 import { 
-    StyleSheet, 
-} from 'react-native';
-import MyMarker from "./MyMarker";
+    Text,
+    TouchableOpacity, 
+    View, 
+    StyleSheet } from "react-native";
+import MapView from 'react-native-maps';
 import BikeMarker from "./BikeMarker";
-import { getBikes } from "../services/bikes";
+import { getBikes, getBikesWithinProximity } from "../services/bikes";
 import { LocationProps } from '../types/types';
+import { useNavigation } from '@react-navigation/native';
+
+const BIKE_RADIUS = 5000;
 
 const Map = (props: LocationProps) => {
 
     const [bikeArray, setBikeArray] = useState([]);
 
+    const navigation = useNavigation();
+
     const userLocation = {
         latitude: props.latitude,
         longitude: props.longitude
     }
 
     useEffect(() => {
-        getBikes()
+        getBikesWithinProximity(BIKE_RADIUS, userLocation)
             .then(data => setBikeArray(data))
             .catch(err => console.log(err));
     }, []);
 
     return (
-        <MapView 
-            style={styles.map}
-            region={{
-                latitude: props.latitude,
-                longitude: props.longitude,
-                latitudeDelta: 0.0222,
-                longitudeDelta: 0.0221,
-                }} 
-            showsUserLocation={true}
-        >
-            {bikeArray.map((bike) => {
-                return <BikeMarker 
-                            bike={bike} 
-                            key={bike.bike_id}
-                            userLocation={userLocation} />
-            })}
-        </MapView>
+        <View style={styles.mapContainer}>
+            <MapView 
+                style={styles.map}
+                region={{
+                    latitude: props.latitude,
+                    longitude: props.longitude,
+                    latitudeDelta: 0.0222,
+                    longitudeDelta: 0.0221,
+                    }} 
+                showsUserLocation={true}
+            >
+                {bikeArray.map((bike) => {
+                    return <BikeMarker 
+                                bike={bike} 
+                                key={bike.bike_id}
+                                userLocation={userLocation} />
+                })}
+            </MapView>
+            <TouchableOpacity 
+                style={styles.listButton}
+                // Had to apply type 'never' to string param for navigate.
+                // Seems like a weird React/TypeScript issue and this is
+                // a quick workaround as found here:
+                // https://stackoverflow.com/questions/68667766/react-native-typescript-string-is-not-assignable-to-parameter-of-type-never
```

Comment:
Wow that's interesting!

Reply:
Yeah, I think it's a little hack-y but it works.